report/3.Theory/ObjectRecognition.tex

```diff
@@ -0,0 +1,146 @@
+
+CNNs are often used along with pooling \cite{CNN}.
+Pooling is a concept in which multiple outputs are combined into a single neuron in the next layer.
+This can be done in order to down-sample the input and reduce the complexity making the CNN run faster
+Some common examples of pooling is max pooling and average pooling, in which the average or max value is output from a number of input neurons.
+
+CNNs are an effective choice for classification. 
+This means it can answer yes or no to whether the image is of a car.
+Now the question is whether it can be used to figure out the position of the objects within the image.
+The answer turns out to be yes.
+
+
+\subsubsection{Regional Convolutional Neural Network}
+
+A regional convolutional neural network (RCNN) utilizes CNN to get both a classification along with a bounding box, which in turn can be used to determine the center of the location.
+It does this by applying the CNN to smaller regions of the image. 
+These regions are extracted from the image using a method called selective search.
+If the region contains an element recognized by the CNN a bounding box can be drawn around the region.
+The bounding box can at last be tightened using different methods, a popular option being a linear regression model.\cite{CNNHistory}
+
+Unfortunately RCNN in itself is not viable for real time object detection.
+However, there have been a lot of modifications and improvements of RCNN that improves the performance of it.
+One improvement is called fast RCNN and is leaning toward being feasible for real time object detection.
+
+%\subsection{You Only Look Once}
+
+%An alternative to RCNN is You Only Look Once (YOLO) which is a new technology developed with real time processing in mind.
+
+
+
+\subsection{GOTURN}
+% https://www.learnopencv.com/goturn-deep-learning-based-object-tracking/
+Generic Object Tracking Using Regression Networks, is an algorithm based on deep learning.\cite{goturn}
+GOTURN is different than the majority of algorithms presented in this section, in that it uses offline learning, whereas online learning constantly improves the model during runtime.
+It is fed a large library of videos of different objects and is taught with the help of this.
+The model is then shipped with the algorithm implementation.
+The benefit of this, is that it runs faster, as it does not have to ever modify the model or analyze whether the given result is relevant.
+However, this comes at a cost. 
+The algorithm has a tendency to favor objects within the training set, even if they remain stationary.
+So, for instance, if we want to track a bird flying past stationary planes, it might lock onto the planes rather than the bird, even though the planes are not moving.
+This is due to the unpredictable-ness of what the algorithm see as a correlation between video and a correct answer.
+
+
+The GOTURN algorithm utilizes CNN, which is explained in the previous sub-section.
+\begin{figure}[H]
```

Comment:
Samme igen, hvad med at bruge denne i stedet for:
\figur{0.9}{images/GOTURN-architecture.jpg}{How the algorithm utilizes CNN, source \cite{goturn}}{fig:goturn-arch}